x/pool-incentives/keeper/grpc_query_test.go

```diff
@@ -35,89 +30,116 @@ func (suite *KeeperTestSuite) TestGaugeIds() {
 	pool, err := suite.App.GAMMKeeper.GetPoolAndPoke(suite.Ctx, poolId)
 	suite.NoError(err)
 
-	res, err := queryClient.GaugeIds(context.Background(), &types.QueryGaugeIdsRequest{
-		PoolId: poolId,
-	})
-	suite.NoError(err)
-	suite.Equal(3, len(res.GaugeIdsWithDuration))
-	suite.Equal(lockableDurations[0], res.GaugeIdsWithDuration[0].Duration)
-	suite.Equal(lockableDurations[1], res.GaugeIdsWithDuration[1].Duration)
-	suite.Equal(lockableDurations[2], res.GaugeIdsWithDuration[2].Duration)
-
-	poolLpDenom := gammtypes.GetPoolShareDenom(pool.GetId())
-	gauge, err := suite.App.IncentivesKeeper.GetGaugeByID(suite.Ctx, res.GaugeIdsWithDuration[0].GaugeId)
-	suite.NoError(err)
-	suite.Equal(0, len(gauge.Coins))
-	suite.Equal(true, gauge.IsPerpetual)
-	suite.Equal(poolLpDenom, gauge.DistributeTo.Denom)
-	suite.Equal(lockableDurations[0], gauge.DistributeTo.Duration)
-
-	gauge, err = suite.App.IncentivesKeeper.GetGaugeByID(suite.Ctx, res.GaugeIdsWithDuration[1].GaugeId)
-	suite.NoError(err)
-	suite.Equal(0, len(gauge.Coins))
-	suite.Equal(true, gauge.IsPerpetual)
-	suite.Equal(poolLpDenom, gauge.DistributeTo.Denom)
-	suite.Equal(lockableDurations[1], gauge.DistributeTo.Duration)
-
-	gauge, err = suite.App.IncentivesKeeper.GetGaugeByID(suite.Ctx, res.GaugeIdsWithDuration[2].GaugeId)
-	suite.NoError(err)
-	suite.Equal(0, len(gauge.Coins))
-	suite.Equal(true, gauge.IsPerpetual)
-	suite.Equal(poolLpDenom, gauge.DistributeTo.Denom)
-	suite.Equal(lockableDurations[2], gauge.DistributeTo.Duration)
+	for _, tc := range []struct {
+		desc    string
+		request *types.QueryGaugeIdsRequest
+		err     bool
+	}{
+		{
+			desc:    "Empty request",
+			request: &types.QueryGaugeIdsRequest{},
+			err:     true,
+		},
+		{
+			desc: "Unexisted pool",
+			request: &types.QueryGaugeIdsRequest{
+				PoolId: 2,
+			},
+			err: true,
+		},
+		{
+			desc: "Happy case",
+			request: &types.QueryGaugeIdsRequest{
+				PoolId: poolId,
+			},
+			err: false,
+		},
+	} {
+		tc := tc
+		suite.Run(tc.desc, func() {
+			res, err := queryClient.GaugeIds(context.Background(), tc.request)
+			if tc.err {
+				suite.Require().Error(err)
+			} else {
+				suite.Require().NoError(err)
+				suite.NoError(err)
+				suite.Equal(3, len(res.GaugeIdsWithDuration))
+
+				poolLpDenom := gammtypes.GetPoolShareDenom(pool.GetId())
+				for i := 0; i < len(res.GaugeIdsWithDuration); i++ {
+					suite.Equal(lockableDurations[i], res.GaugeIdsWithDuration[i].Duration)
+
+					gauge, err := suite.App.IncentivesKeeper.GetGaugeByID(suite.Ctx, res.GaugeIdsWithDuration[i].GaugeId)
+					suite.NoError(err)
+					suite.Equal(0, len(gauge.Coins))
```

Comment:
nit: I think the verdict was to use `suite.Require()` whenever necessary [link](https://github.com/osmosis-labs/osmosis/pull/2046#pullrequestreview-1041071475)

Reply:
yup!!!